dataflux_pytorch/dataflux_mapstyle_dataset.py

```diff
@@ -140,11 +139,18 @@ def __init__(
         self.config = config
         # If composed download is enabled, check if the client has permissions to create and delete the composed object.
         if self.config.max_composite_object_size != 0:
-            missing_perm = _get_missing_permissions(
-                storage_client=self.storage_client,
-                bucket_name=self.bucket_name,
-                project_name=self.project_name,
-                required_perm=[CREATE, DELETE])
+            try:
+                missing_perm = _get_missing_permissions(
+                    storage_client=self.storage_client,
+                    bucket_name=self.bucket_name,
+                    project_name=self.project_name,
+                    required_perm=[CREATE, DELETE])
+            except RefreshError as e:
+                e.add_note(
+                    "Default credentials might be missing. Run `gcloud auth application-default login`")
```

Comment:
Same comment here as above.

Also if we are going to add this note in everywhere it occurs, why not add it just in _helper.py instead of both dataflux_iterable_dataset.py and dataflux_mapstyle_dataset.py?